spec/formatting.md

```diff
@@ -18,3 +18,236 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
```

Comment:
Nitpick / editorial:
How do you feel about having a small list with titles only, almost like a TOC
Followed by the details, with examples.

It would help with the "big picture", vs something spread over many paragraphs.
```
These are divided into the following categories:
* Syntax errors
* Data Model errors
    * Variant Key Mismatch errors
    * Missing Fallback Variant errors
* Resolution errors
    * Unresolved Variable errors
    * Unknown Function errors
* Selection errors
    * Selector errors
* Formatting errors
```

Reply:
That's how this started out, before the examples got added...

Let's revisit this when we've more of the spec gathered up? The solution for this section is likely to look very similar to other sections as well.